### DISCUSSION TITLE
Why do we need to chown the /run/docker.sock to make docker operator work ?

### DISCUSSION THREAD

**Discussion by ginolegigot** [apache/airflow#34757] [Q&A]
Hello!
I'm in a celery cluster 2 nodes with scheduler webserver rabbitMQ processes and two worker nodes with celery worker and flower processes (and docker installed on these nodes).
I tried to trigger a job through Airflow UI, but i encountered an error about docker socket permission error.
```shell
Traceback (most recent call last):
  File "/opt/airflow/lib/python3.10/site-packages/urllib3/connectionpool.py", line 714, in urlopen
    httplib_response = self._make_request(
  File "/opt/airflow/lib/python3.10/site-packages/urllib3/connectionpool.py", line 415, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/opt/airflow/lib/python3.10/site-packages/urllib3/connection.py", line 244, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/lib/python3.10/http/client.py", line 1283, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1329, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1278, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1038, in _send_output
    self.send(msg)
  File "/usr/lib/python3.10/http/client.py", line 976, in send
    self.connect()
  File "/opt/airflow/lib/python3.10/site-packages/docker/transport/unixconn.py", line 27, in connect
    sock.connect(self.unix_socket)
PermissionError: [Errno 13] Permission denied
```
I started a journey through many stackoverflow threads, saying all basically the same thing, enable airflow permissions on this docker socket. I did not find this as a satisfying answer so i wanted to reproduce the issue through this command (run as airflow user which is inside the docker group like a normal user wishing to use docker):
```shell
airflow dags test docker_dag
```
But instead this dag run was successful despite permission as root/docker on the /run/docker.sock.
So here my question, if i run a dag using a docker_operator with traditional rights on the /run/docker.sock (meaning root/docker) why does it work through airflow dags cli (as airflow user) but does not work through a webserver (UI) manual or scheduled trigger ?
Does it mean running a dag through UI uses another user than airflow ?

**Comment by Taragolis** [comment]
There is no much details by I asume that you run your Airflow cluster in docker, and this refers to how how [DinD](https://www.google.com/search?q=docker+in+docker) (Docker in Docker) works

Run Docker-in-Docker could have different issues time to time, and depend on multiple factors Host OS, docker version and etc.
And it might required to use different flags, see:
- https://github.com/docker-library/docker/issues/151#issuecomment-483185972
- https://github.com/docker-library/docker/issues/281#issuecomment-744766015
- https://github.com/moby/moby/pull/40759

**Reply by ginolegigot** [reply]
sorry, running into a celery cluster of VM

**Reply by ginolegigot** [reply]
I rephrased a bit for more clarity
